src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -435,14 +367,74 @@ namespace AppInstaller::CLI::Workflow
                 context.Reporter.Warn() << Resource::String::OverwritingExistingFileAtMessage << ' ' << symlinkFullPath.u8string() << std::endl;
             }
 
-            std::filesystem::create_symlink(targetFullPath, symlinkFullPath);
-            AICLI_LOG(CLI, Info, << "Symlink created at: " << symlinkFullPath);
-            uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, symlinkFullPath.wstring());
+            if (Filesystem::CreateSymlink(targetFullPath, symlinkFullPath))
+            {
+                AICLI_LOG(CLI, Info, << "Symlink created at: " << symlinkFullPath);
+                uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, symlinkFullPath.wstring());
+            }
+            else
+            {
+                // Symlink creation should only fail if the user executes in user mode and non-admin.
+                // Resort to adding install directory to PATH directly.
+                AICLI_LOG(CLI, Info, << "Portable install executed in user mode. Adding package directory to PATH.");
+                bool installDirectoryAddedToPath = true;
+                uninstallEntry.SetValue(PortableValueName::InstallDirectoryAddedToPath, installDirectoryAddedToPath);
+            }
+        }
 
-            Manifest::ScopeEnum scope = ConvertToScopeEnum(context.Args.GetArg(Execution::Args::Type::InstallScope));
-            if (AddToPathRegistry(scope))
+        void AddToPathVariable(Execution::Context& context)
+        {
+            PortableARPEntry& uninstallEntry = context.Get<Execution::Data::PortableARPEntry>();
+            Manifest::ScopeEnum scope = uninstallEntry.GetScope();
+            const auto& installDirectoryAddedToPath = uninstallEntry[PortableValueName::InstallDirectoryAddedToPath];
```

Comment:
You shouldn't be passing information from one task to another through the registry.

Suppose:
1. Install 1.0 and it can't create a symlink, so it adds to path
2. Upgrade to 2.0 as admin so it can create a symlink
3. This code sees that it should add to path again (although I think you do check for the path being there already)
4. Oops, now both a symlink and a path?

This is one reason not to pass the information using the registry, but it also brings up an interesting question of what to do on upgrades around this behavior.  I would think my case above should just preserve the path and not create a symlink.  I would think that the inverse (admin then user) should attempt to preserve the symlink, as if it doesn't need to be changed (same link and target paths) then there is no need.

Basically, we should determine if any changes are actually needed, then only attempt to make them if necessary.

My intuition says that this would be easier to understand if everything was collected in an object structure rather than having every task handling many things repeatedly.